src/fam.rs

```diff
@@ -938,4 +959,55 @@ mod tests {
         type Message2FamStructWrapper = FamStructWrapper<Message2>;
         assert!(serde_json::from_str::<Message2FamStructWrapper>(data_ser.as_str()).is_err());
     }
+
+    #[test]
+    fn test_clone_multiple_fields() {
+        #[derive(Default)]
+        #[repr(C)]
+        struct Foo {
+            index: u32,
+            length: u16,
+            flags: u32,
+            entries: __IncompleteArrayField<u32>,
+        }
+
+        generate_fam_struct_impl!(Foo, u32, entries, u16, length, 100);
+
+        type FooFamStructWrapper = FamStructWrapper<Foo>;
+
+        let mut foo = FooFamStructWrapper::new(0).unwrap();
+        foo.as_mut_fam_struct().index = 1;
+        foo.as_mut_fam_struct().flags = 2;
+        foo.as_mut_fam_struct().length = 3;
+        foo.push(3).unwrap();
+        foo.push(14).unwrap();
+        assert_eq!(foo.as_slice().len(), 3 + 2);
+        assert_eq!(foo.as_slice()[3 + 0], 3);
+        assert_eq!(foo.as_slice()[3 + 1], 14);
+
+        let mut foo2 = foo.clone();
+        assert_eq!(
+            foo.as_mut_fam_struct().index,
+            foo2.as_mut_fam_struct().index
+        );
+        assert_eq!(
+            foo.as_mut_fam_struct().length,
+            foo2.as_mut_fam_struct().length
+        );
+        assert_eq!(
+            foo.as_mut_fam_struct().flags,
+            foo2.as_mut_fam_struct().flags
+        );
+        assert_eq!(foo.as_slice(), foo2.as_slice());
+        assert_eq!(foo2.as_slice().len(), foo2.as_mut_fam_struct().length as usize);
```

Comment:
This also confuses me..

`foo.as_slice().len()` was checked to be `3 + 2`, then here `foo2.as_slice().len()` is checked to be `3` (`foo2.as_mut_fam_struct().length` is 3).